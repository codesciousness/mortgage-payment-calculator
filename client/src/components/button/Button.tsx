import React from 'react';
import LoadingButton from '@mui/lab/LoadingButton';
import SendIcon from '@mui/icons-material/Send';

type ButtonProps = {
    name: string;
    loading?: boolean;
    onClick: () => void;
};

const Button = ({ name, loading, onClick }: ButtonProps): JSX.Element => {
    const id = name.replaceAll(' ', '');

    return (
        <LoadingButton
            id={id}
            variant='outlined'
            loading={loading}
            endIcon={<SendIcon />}
            loadingPosition="end"
            onClick={onClick}
            sx={{
                fontWeight: 'bold',
                margin: '1rem',
                color: 'DeepSkyBlue',
                background: 'linear-gradient(45deg, white, white, white, PaleTurquoise, PowderBlue, SkyBlue, DeepSkyBlue)',
                backgroundSize: '300% 100%',
                transition: 'all 1s ease-in-out',
                '&:hover': {
                    color: 'white',
                    backgroundPosition: '100% 0'
                }
            }}
        >
            {name}
        </LoadingButton>
    );
};

export default Button;