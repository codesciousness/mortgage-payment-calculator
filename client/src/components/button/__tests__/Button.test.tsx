import React from 'react';
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import Button from '../Button';

const name = 'Button Name';
const width = 300;
const loadingTrue = true;
const loadingFalse = false;
const onClick = () => null;

describe('<Button />', () => {
    test('Button component is defined', () => {
        expect(<Button name={name} onClick={onClick}/>).toBeDefined();
    });
    test('component matches the snapshot', () => {
        const component = create(<Button name={name} width={width} loading={loadingFalse} onClick={onClick}/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('renders correctly when loading prop is true', () => {
        render(<Button name={name} width={width} loading={loadingTrue} onClick={onClick}/>);
        expect(screen.getByText('Button Name')).toBeInTheDocument();
    });
    test('renders correctly when loading prop is false', () => {
        render(<Button name={name} width={width} loading={loadingFalse} onClick={onClick}/>);
        expect(screen.getByText('Button Name')).toBeInTheDocument();
    });
});