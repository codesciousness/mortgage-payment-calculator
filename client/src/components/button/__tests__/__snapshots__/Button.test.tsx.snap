// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> component matches the snapshot 1`] = `
<button
  className="MuiLoadingButton-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root css-1s9360s-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
  disabled={false}
  id="ButtonName"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Button Name
  <span
    className="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="SendIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
      />
    </svg>
  </span>
</button>
`;
